src/HttpRequestParser.cpp

```diff
@@ -31,6 +33,12 @@ HttpRequest *HttpRequestParser::parseHttpRequest(std::string requestInput, Serve
 	}
 	request->setHost(request->getHeader("Host"));
 	return request;
+	}
+	catch (...)
+	{
+		delete request;
+		throw;
```

Comment:
throwing what? does this shut down the program? that shouldnt happen

Reply:
we also need to add the specific exception to be catched here.. rather multiple catches that catchs different exceptions and return after the try catch block.

Reply:
Also, the request will be deleted in the error response thing in the server handler so the leak is not the request itself... in anycase it speaks about some deeper issue I think

Reply:
so the ellipses is a catch-all, and the empty throw will throw the same error it received